parse/parse.go

```diff
@@ -39,22 +40,33 @@ func New(name string, env []string, r *Restrictions) *Parser {
 }
 
 // Parse parses the given string.
-func (p *Parser) Parse(text string) (string, error) {
+func (p *Parser) Parse(text string, failOnfirst bool) (string, error) {
```

Comment:
This breaks the existing the API. Let's add a `Mode` field to the parser. Something like [go/parser.Mode](https://golang.org/pkg/go/parser/#Mode).

Reply:
great, will do that soon.

Reply:
@a8m I have now updated it, introduced a test for negative assertions as well. reverted the Parse interface , to original implementation and added `AllErrors`. Let me know if this is sufficient.

Reply:
Thanks for adding this. I like this feature, but I think some of the implementation should be different. 

IMO, the `Mode` field I mentioned above, should be part of the Parser struct and not the Restriction struct since it controls the parser behavior. I also expect it to be extended in the future (for more flags/options).

Reply:
@a8m - sorry for the delay, not had time for the last week. totally see your point, so adding that field to the struct will still need to change envsubst, we could leave it unchanged and deduce it ("Mode") based on existing restrictions but then you could want to "fail-fast" or see all errors in strict `Restriction` 

so if you are OK with extending the StringRestricted and ByteRestrcited to accept a `Mode` then the Parser struct can be initialized with caller specified Mode. 
what do you think?

Reply:
> so if you are OK with extending the StringRestricted and ByteRestrcited to accept a Mode then the Parser can be initialized with caller specified Mode.

I think that "fail-fast" is useful option for the CLI version of envsubst (like the case you've described above).
I prefer to keep the new Mode options only in the `parse` package and leave the `XXXRestricted` functions without changes. 

Besides that, code looks good and can be merged.

Reply:
right, @a8m this should all be as expected now - sorry about the delay was being a bit slow understanding what you meant. Let me know if any more changes required - otherwise could you bump the release if all good your side.